scrapy/pipelines/files.py

```diff
@@ -241,6 +241,38 @@ def media_downloaded(self, response, request, info):
         referer = referer_str(request)
 
         if response.status != 200:
+
+            if response.status == 201:
+
+                try:
+                    loc = [response.headers['location']]
```

Comment:
Ha. I can do away with the `try`, `catch` now. Also, will look into the `getlist(key)` method.